==1528839== Memcheck, a memory error detector
==1528839== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==1528839== Using Valgrind-3.18.1 and LibVEX; rerun with -h for copyright info
==1528839== Command: ./philo 4 410 200 200
==1528839== 
==1528839== 
==1528839== Process terminating with default action of signal 2 (SIGINT)
==1528839==    at 0x494E7F8: clock_nanosleep@@GLIBC_2.17 (clock_nanosleep.c:78)
==1528839==    by 0x4953676: nanosleep (nanosleep.c:25)
==1528839==    by 0x498511E: usleep (usleep.c:31)
==1528839==    by 0x1097BB: monitor_status (in /home/wnguyen/42cursus/git/philo_v3/philo)
==1528839==    by 0x109521: main (in /home/wnguyen/42cursus/git/philo_v3/philo)
==1528839== 
==1528839== HEAP SUMMARY:
==1528839==     in use at exit: 5,856 bytes in 7 blocks
==1528839==   total heap usage: 7 allocs, 0 frees, 5,856 bytes allocated
==1528839== 
==1528839== 160 bytes in 1 blocks are still reachable in loss record 1 of 4
==1528839==    at 0x4848899: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==1528839==    by 0x109F8A: args_init (in /home/wnguyen/42cursus/git/philo_v3/philo)
==1528839==    by 0x10947C: main (in /home/wnguyen/42cursus/git/philo_v3/philo)
==1528839== 
==1528839== 512 bytes in 1 blocks are still reachable in loss record 2 of 4
==1528839==    at 0x4848899: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==1528839==    by 0x109F46: args_init (in /home/wnguyen/42cursus/git/philo_v3/philo)
==1528839==    by 0x10947C: main (in /home/wnguyen/42cursus/git/philo_v3/philo)
==1528839== 
==1528839== 1,088 bytes in 4 blocks are possibly lost in loss record 3 of 4
==1528839==    at 0x484DA83: calloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==1528839==    by 0x40147D9: calloc (rtld-malloc.h:44)
==1528839==    by 0x40147D9: allocate_dtv (dl-tls.c:375)
==1528839==    by 0x40147D9: _dl_allocate_tls (dl-tls.c:634)
==1528839==    by 0x48FE7B4: allocate_stack (allocatestack.c:430)
==1528839==    by 0x48FE7B4: pthread_create@@GLIBC_2.34 (pthread_create.c:647)
==1528839==    by 0x10937C: run_simulation (in /home/wnguyen/42cursus/git/philo_v3/philo)
==1528839==    by 0x1094E6: main (in /home/wnguyen/42cursus/git/philo_v3/philo)
==1528839== 
==1528839== 4,096 bytes in 1 blocks are still reachable in loss record 4 of 4
==1528839==    at 0x4848899: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==1528839==    by 0x48E7BA3: _IO_file_doallocate (filedoalloc.c:101)
==1528839==    by 0x48F6CDF: _IO_doallocbuf (genops.c:347)
==1528839==    by 0x48F5F5F: _IO_file_overflow@@GLIBC_2.2.5 (fileops.c:744)
==1528839==    by 0x48F46D4: _IO_new_file_xsputn (fileops.c:1243)
==1528839==    by 0x48F46D4: _IO_file_xsputn@@GLIBC_2.2.5 (fileops.c:1196)
==1528839==    by 0x48DECA8: outstring_func (vfprintf-internal.c:239)
==1528839==    by 0x48DECA8: __vfprintf_internal (vfprintf-internal.c:1516)
==1528839==    by 0x48C979E: printf (printf.c:33)
==1528839==    by 0x10A118: print_status (in /home/wnguyen/42cursus/git/philo_v3/philo)
==1528839==    by 0x10A2D9: pickup_forks (in /home/wnguyen/42cursus/git/philo_v3/philo)
==1528839==    by 0x10A421: eat (in /home/wnguyen/42cursus/git/philo_v3/philo)
==1528839==    by 0x10A794: philo_routine (in /home/wnguyen/42cursus/git/philo_v3/philo)
==1528839==    by 0x48FDAC2: start_thread (pthread_create.c:442)
==1528839== 
==1528839== LEAK SUMMARY:
==1528839==    definitely lost: 0 bytes in 0 blocks
==1528839==    indirectly lost: 0 bytes in 0 blocks
==1528839==      possibly lost: 1,088 bytes in 4 blocks
==1528839==    still reachable: 4,768 bytes in 3 blocks
==1528839==         suppressed: 0 bytes in 0 blocks
==1528839== 
==1528839== ERROR SUMMARY: 1 errors from 1 contexts (suppressed: 0 from 0)
